DEV_GUIDE.md

```diff
@@ -0,0 +1,166 @@
+#Developer guide
+
+##Contributing new content
+
+In addition to including features / fixes, you can contribute new content to Scala Exercises! The platform is designed in a way that creating new exercises is pretty straight-forward. Currently there are several libraries available in production (i.e.: STD lib, Cats, Shapeless...), and more are to be included soon. If you're interested in creating a new exercises library, you'll find all the needed steps in this document.
+
+###Overview
+
+The `scala-exercises` organization holds different repositories for both the core project ("scala-exercises"), and each of the main exercises libraries (i.e.: "exercises-stdlib", "exercises-cats"...). The core project is currently built on one single module. The content you can provide to the platform doesn't necessarily be in the previously mentioned organization, but for them to appear inside the main `scala-exercises` in production, they need to be published in a public artifact repository like Sonatype or Maven-central. All the steps, from creating the content library to the publication process, are described in the following sections.
+
+###Content creation
+
+Scala Exercises content is organized in the following way:
+
+```
+Scala Exercises Core
+       |
+       | imports
+       |
+   Libraries
+       |
+       | contains
+       |
+    Sections
+       |
+       | contains
+       |
+    Exercises
+```
+
+####Creating a new content library
+
+Each content library is an independent SBT module working as a library, which should be published as an artifact for it to be included in Scala Exercises. From now on, we'll consider you've already set up a new project for your exercises library, and we'll focus on the specifics on how to layout the content itself.
+
+First of all you need to set up the library by creating a new Scala `object` that contains basic details about it. This `object` must subclass from `org.scalaexercises.definitions.Library`. Here you can find a template for this object:
+
+```scala
+package package_name (1)
+
+import org.scalaexercises.definitions._
+
+/** This line should describe your library shortly, and will appear as its description in the main Scala Exercises website. (2)
+  *
+  * @param name library_name (3)
+  */
+object MyLibrary extends Library {
+  override def owner = "owner" (4)
+  override def repository = "github_repository" (5)
+
+  override def color = Some("#HEX_COLOR") (6)
+
+  (7)
+  override def sections = List(
+    SectionA,
```

Comment:
Might be slightly confusing: called `SectionA` here, but `MySection` below. Also, you may need to import or otherwise reference the sections?